homeassistant/components/switchbee/coordinator.py

```diff
@@ -15,7 +31,26 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class SwitchBeeCoordinator(DataUpdateCoordinator[dict[int, SwitchBeeBaseDevice]]):
+class SwitchBeeCoordinator(
+    DataUpdateCoordinator[
+        dict[
+            int,
+            Union[
+                SwitchBeeSwitch,
+                SwitchBeeGroupSwitch,
+                SwitchBeeTimedSwitch,
+                SwitchBeeShutter,
+                SwitchBeeSomfy,
+                SwitchBeeDimmer,
+                SwitchBeeThermostat,
+                SwitchBeeScenario,
+                SwitchBeeRollingScenario,
+                SwitchBeeTimerSwitch,
+                SwitchBeeTwoWay,
+            ],
+        ]
+    ]
+):
```

Comment:
I think using SwitchBeeBaseDevice is a lot clearer.
Since dict is now causing issues with the library, you can instead return Mapping (`from collections.abc import Mapping`)
```suggestion
class SwitchBeeCoordinator(DataUpdateCoordinator[Mapping[int, SwitchBeeBaseDevice]]):
```